### PR TITLE
fix: [PL-23534]: git msvc will call other msvc in sorted order

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)

### CODE REVIEW

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java, line 123 at r1](https://reviewable.io/reviews/harness/harness-core/31317#-Mxji0Kl4LF_yAlYWwCL:-Mxji0Kl4LF_yAlYWwCM:b-vmvyld) ([raw file](https://github.com/harness/harness-core/blob/366e803901160f54f45acd7b10a056c275b838b5/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L123)):*
> ```Java
>   @Singleton
>   List<Microservice> getMicroservicesProcessingOrder() {
>     return Arrays.asList(CORE, TEMPLATESERVICE, PMS, CF, POLICYMGMT);
> ```

Check with pms whether policy management is required before pms

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static GitSyncModule getInstance() {
   @Provides
   @Singleton
   List<Microservice> getMicroservicesProcessingOrder() {
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java:122]: CF should be ahead imo

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 4 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[900-git-sync-sdk/src/main/java/io/harness/gitsync/utils/GitSdkUtils.java, line 13 at r2](https://reviewable.io/reviews/harness/harness-core/31317#-Mxo2dEy1QVWbommGtf2:-Mxo2dEy1QVWbommGtf3:bfdf06v) ([raw file](https://github.com/harness/harness-core/blob/67a2ad56bd41a1dc3004320259fd4e50cf1f1104/900-git-sync-sdk/src/main/java/io/harness/gitsync/utils/GitSdkUtils.java#L13)):*
> ```Java
> @UtilityClass
> @OwnedBy(PL)
> public class GitSdkUtils {
> ```

The name is quite generic, we can put this function in gitcontext helper too

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 6 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java, line 122 at r1](https://reviewable.io/reviews/harness/harness-core/31317#-MxjYIOj8_xrqtxE90zz-r1-122:-MxoDNzb80VdebJ6inzp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8e96f4e8d07fb54adb67afa0b3f6a81670f1bf02/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L122)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

CF should be ahead imo
</blockquote></details>

Done.
___
*[900-git-sync-sdk/src/main/java/io/harness/gitsync/utils/GitSdkUtils.java, line 13 at r2](https://reviewable.io/reviews/harness/harness-core/31317#-Mxo2dEy1QVWbommGtf2:-MxoDP8x-gF-XlClaISc:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/67a2ad56bd41a1dc3004320259fd4e50cf1f1104/900-git-sync-sdk/src/main/java/io/harness/gitsync/utils/GitSdkUtils.java#L13)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

The name is quite generic, we can put this function in gitcontext helper too
</blockquote></details>

Done.

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 6 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 2 files at r2, 2 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 2 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31317)* status: :shipit: complete! ✓

**Pull request merged**